frontend/src/views/DiagnosticTunnel/QualityMeasureSteps/FamilyFieldsStep.vue

```diff
@@ -117,47 +150,154 @@ export default {
+          this.outsideLawErrorMessages.push(message)
+        }
+      }
+
+      // check meat and fish totals
+      const { meatPoultryEgalim } = this.meatPoultryTotals()
+      const sumMeat = meatPoultryEgalim + (this.payload.valueViandesVolaillesNonEgalim || 0)
+      const totalMeat = this.payload.valueMeatPoultryHt
+      if (sumMeat > totalMeat) {
+        this.meatTotalErrorMessage = this.errorMessage(sumMeat, totalMeat, 3, this.meatFieldPrefix)
+      }
+      const { fishEgalim } = this.fishTotals()
+      const sumFish = fishEgalim + (this.payload.valueProduitsDeLaMerNonEgalim || 0)
+      const totalFish = this.payload.valueFishHt
+      if (sumFish > totalFish) {
+        this.fishTotalErrorMessage = this.errorMessage(sumFish, totalFish, 3, this.fishFieldPrefix)
+      }
+
+      if (!this.hasError) {
+        this.errorOnLoad = false
+        this.erroringFieldName = null
+      }
+    },
+    getOutsideLawTotalErrorMessage(fieldSuffix) {
+      const outsideLaw = Constants.TeledeclarationCharacteristicGroups.outsideLaw
+      const fields = outsideLaw.fields.filter((field) => field.endsWith(fieldSuffix))
+      const total = this.sum(fields)
+      if (total > this.payload.valueTotalHt) {
+        const char = getCharacteristicFromFieldSuffix(fieldSuffix, outsideLaw)
+        return this.errorMessage(total, this.payload.valueTotalHt, 1, undefined, char.text)
+      }
+    },
+    fieldUpdate(fieldName) {
+      if (this.diagnostic[fieldName] === this.payload[fieldName]) return
+      this.checkTotal()
+      this.populateSimplifiedDiagnostic()
+    },
+    sum(fields) {
+      return fields.reduce((acc, field) => acc + (this.payload[field] || 0), 0)
+    },
+    fieldHasError(fieldName) {
+      if (fieldName.startsWith(this.meatFieldPrefix) && !!this.meatTotalErrorMessage) return true
+      if (fieldName.startsWith(this.fishFieldPrefix) && !!this.fishTotalErrorMessage) return true
+      return !!this.payload[fieldName] && (this.errorOnLoad || this.hasTotalError)
+    },
+    errorMessage(problemValue, totalValue, stepNumber, familyFieldPrefix, characteristicText) {
+      // family can be undefined
+      const familyTextChoices = {
+        [this.fishFieldPrefix]: "des produits aquatiques ",
+        [this.meatFieldPrefix]: "des viandes et volailles ",
+      }
+      let familyText = familyTextChoices[familyFieldPrefix] || ""
+      const stepTextChoices = {
+        1: "dans la première étape ",
+        3: "dans la troisième étape ",
+      }
+      const stepText = stepTextChoices[stepNumber] || ""
+      problemValue = toCurrency(problemValue)
+      totalValue = toCurrency(totalValue)
+      if (characteristicText) {
+        characteristicText = `« ${characteristicText} » `
+        return `Le total ${familyText}${characteristicText}(${problemValue}) excédent le total ${familyText}saisi ${stepText}(${totalValue})`
```

Comment:
Dans ce cas on parle d'un seul champ non ? (ou d'un seul _total_ ?) dans ce cas on peut conjuguer à la troisième personne
 ```suggestion
        return `Le total ${familyText}${characteristicText}(${problemValue}) excède le total ${familyText}saisi ${stepText}(${totalValue})`
```